community/kernel/src/main/java/org/neo4j/kernel/api/proc/ProcedureSignature.java

```diff
@@ -174,14 +177,26 @@ public int hashCode()
     }
 
     public ProcedureSignature( ProcedureName name,
-                               List<FieldSignature> inputSignature,
-                               List<FieldSignature> outputSignature,
-                               Mode mode )
+            List<FieldSignature> inputSignature,
+            List<FieldSignature> outputSignature,
+            Mode mode,
+            boolean deprecated,
```

Comment:
Interesting. Have not seen idiomatic use of Optional in Java. Could consider that.